sdk/tables/Azure.Data.Tables/src/Queryable/ReflectionUtil.cs

```diff
@@ -23,16 +24,18 @@ internal static class ReflectionUtil
         static ReflectionUtil()
         {
             DictionaryGetItemMethodInfo = typeof(IDictionary<string, object>).GetMethod("get_Item");
-            DynamicDictionaryGetBinaryMethodInfo = typeof(DynamicTableEntity).GetMethod(nameof(DynamicTableEntity.GetBinary));
-            DynamicDictionaryGetBooleanMethodInfo = typeof(DynamicTableEntity).GetMethod(nameof(DynamicTableEntity.GetBoolean));
-            DynamicDictionaryGetDateTimeMethodInfo = typeof(DynamicTableEntity).GetMethod(nameof(DynamicTableEntity.GetDateTime));
-            DynamicDictionaryGetDoubleMethodInfo = typeof(DynamicTableEntity).GetMethod(nameof(DynamicTableEntity.GetDouble));
-            DynamicDictionaryGetGuidMethodInfo = typeof(DynamicTableEntity).GetMethod(nameof(DynamicTableEntity.GetGuid));
-            DynamicDictionaryGetInt32MethodInfo = typeof(DynamicTableEntity).GetMethod(nameof(DynamicTableEntity.GetInt32));
-            DynamicDictionaryGetInt64MethodInfo = typeof(DynamicTableEntity).GetMethod(nameof(DynamicTableEntity.GetInt64));
-            DynamicDictionaryGetString64MethodInfo = typeof(DynamicTableEntity).GetMethod(nameof(DynamicTableEntity.GetString));
+            DynamicDictionaryGetItemMethodInfo = typeof(TableEntity).GetMethod("get_Item");
```

Comment:
@christothes Just in case, I wanted to make sure that this was okay! I added a separate MethodInfo for the `TableEntity` indexer get because the one above for `IDictionary` wasn't working. I wasn't sure if there was a different way that wouldn't require explicitly calling `TableEntity`.

Reply:
I think it's fine